### PR TITLE
feat(ForumChannel): add basic implementation

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

👋 Long time no see.

This PR adds **a basic** implementation of Forum Channels, since there are some things (such as tags) that are likely subject to change, however, I kept the types.

Forum channels are very similar to Text Channels, that's why you'll see a lot of repeated methods/properties.

#### Checklist

- Implement Forum Channels
  - [x] Deserialization.
  - [ ] Implement `available_tags` field. (see below)
  - [ ] Edit method. (To be done and can't test atm since the API is locked to bots)
  - [ ] Create thread method. (It's done, but I can't test)
  - [x] Archived threads method.
- Changes in Thread
  - [x] Implement `flags` field (and thus `ChannelFlags` class)
  - [ ] Implement `applied_tags` field. (see below)
  - [ ] Implement Tags (I already have an implementation, but I deleted it since it's subject to change)
- Changes in HTTPClient
   - [x] `create_forum_post` (Rename to `create_forum_thread`?) - `POST /channels/:id/threads`
   - [x] `create_tag` (Not used) - `POST /channels/:id/tags`
   - [x] `edit_tag` (Not used) - `PUT /channels/:id/tags/:id`
   - [x] `delete_tag` (Not used) - `DELETE /channels/:id/tags/:id`
- Changes in Audit Log
   note: There are no audit logs when a Tag is created/updated/deleted (yet?)
   - [ ] `applied_tags` key, when adding/deleting a tag to a thread
- Types
  - [x] Forum Channel type.
  - [x] Tag type.
  - [x] Add `flags` and `applied_tags` to thread type. 

#### References
- https://github.com/discord-payloads/discord-payloads/pull/3
- https://gist.github.com/Victorsitou/913e741270e0cc1f3cdccbb04210e24e
- https://github.com/discord/discord-api-docs/pull/4761

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
> [last_message_id -> last_thread_id](https://github.com/DisnakeDev/disnake/pull/448/commits/113282b5d3b169a47965b5fbea65e808277df42b)

Should we keep this named last_message_id and have an alias of last_thread_id?

File: disnake/channel.py [hunk 1]
```diff
@@ -1806,6 +1814,491 @@ class NewsChannel(TextChannel):
...
+
+    def archived_threads(
+        self,
+        *,
+        private: bool = False,
+        joined: bool = False,
+        limit: Optional[int] = 50,
+        before: Optional[Union[Snowflake, datetime.datetime]] = None,
+    ) -> ArchivedThreadIterator:
+        """Returns an :class:`~disnake.AsyncIterator` that iterates over all archived threads in the guild.
```
Comment (onerandomusername) [inline disnake/channel.py:2264]: ``in the channel.``

Wherever this was copied from (I think TextChannel) should also be changed.

File: disnake/http.py [hunk 2]
```diff
@@ -1174,6 +1175,80 @@ def get_thread_members(self, channel_id: Snowflake) -> Response[List[threads.Thr
...
+    ) -> Response[threads.Thread]:
+        valid_keys = (
+            # Post fields
+            "name",
+            "auto_archive_duration",
+            "rate_limit_per_user",
+            # Message fields
+            "content",
+            "embeds",
+            "embed",
```
Comment (onerandomusername) [inline disnake/http.py:1235]: Not actually a valid key, this needs to be processed beforehand.

What we've done in the past is set `embeds = [embed]`

File: disnake/http.py [hunk 3]
```diff
@@ -1174,6 +1175,80 @@ def get_thread_members(self, channel_id: Snowflake) -> Response[List[threads.Thr
...
+            "attachments",
+            "flags",
+        )
+        payload = {k: v for k, v in fields.items() if k in valid_keys}
+        payload["type"] = 11
+        route = Route("POST", "/channels/{channel_id}/threads", channel_id=channel_id)
+
+        if files is not None:
+            multipart = to_multipart_with_attachments(payload, files)
+            return self.request(route, form=multipart, files=files, reason=reason)
```
Comment (onerandomusername) [inline disnake/http.py:1248]: Please include a param 'has_message' to True, per https://canary.discord.com/channels/927603005627437097/927603006579568723/962075735835672686

File: disnake/threads.py [hunk 4]
```diff
@@ -371,6 +380,17 @@ def is_nsfw(self) -> bool:
         parent = self.parent
         return parent is not None and parent.is_nsfw()
 
+    def is_pinned(self) -> bool:
+        """Whether the thread is pinned in a :class:`ForumChannel`
+
+        Pinned threads do not get affected by the auto archive duration.
```
Comment (onerandomusername) [inline disnake/threads.py:386]: are not affected

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/threads.py (hunk 4 above)
Comment (Victorsitou) [reply disnake/threads.py:386]: > A thread can be pinned within a forum. A thread that is pinned will have the `(1 << 1)` flag set. Archiving a pinned thread will unset the flag. A pinned thread will not auto archive.

https://github.com/discord/discord-api-docs/commit/fba22cbd77f537c0d6efaa1b800ba58952a4355a#diff-e28250ee18796453c6332a572fb2fe1f25be18b46bbb56934789cfaef57edec2R193

File: disnake/channel.py (hunk 1 above)
Comment (Victorsitou) [reply disnake/channel.py:2264]: 😓😓

File: disnake/threads.py (hunk 4 above)
Comment (onerandomusername) [reply disnake/threads.py:386]: I meant we should change this line to `Pinned threads are not affected by the auto archive duration.`

File: disnake/channel.py [hunk 5]
```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
...
+        You must have the :attr:`~Permissions.send_messages` permission to do this.
+
+        All the things that are in :meth:`TextChannel.send` are also valid here.
+
+        .. danger::
+
+            This method does two API calls, one for creating the thread and one for
+            sending the message. Please be aware of this to avoid doing a lot of
+            API calls.
+
```
Comment (onerandomusername) [inline disnake/channel.py:2183]: No longer true! Discord fixed the endpoint!

File: disnake/channel.py [hunk 6]
```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
...
+            You do not have permission to fetch this channel.#
+
+        Returns
+        -------
+        :class:`.Thread`
+            The channel from the ID.
+        """
+        return await self.guild.fetch_channel(thread_id)  # type: ignore
+
+    async def create_thread(
```
Comment (onerandomusername) [inline disnake/channel.py:2153]: perhaps this should be named `create_post`?

File: disnake/channel.py [hunk 7]
```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
...
+        self,
+        *,
+        name: str,
+        auto_archive_duration: AnyThreadArchiveDuration = MISSING,
+        slowmode_delay: int = MISSING,
+        content: Any,
```
Comment (onerandomusername) [inline disnake/channel.py:2159]:
```suggestion
        content: str,
```
(per #411)

File: disnake/channel.py [hunk 8]
```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
...
+            You do not have permissions to create a thread.
+        HTTPException
+            Starting the thread failed.
+
+        Returns
+        -------
+        :class:`Thread`
+            The newly created thread.
+        """
+        from .webhook.async_ import handle_message_parameters
```
Comment (onerandomusername) [inline disnake/channel.py:2239]: This method should probably be defined in http.py instead of the webhooks

File: disnake/channel.py [hunk 9]
```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
...
+        self.default_auto_archive_duration: ThreadArchiveDurationLiteral = data.get(
+            "default_auto_archive_duration", 1440
+        )
+        self._fill_overwrites(data)
+
+    async def _get_channel(self) -> ForumChannel:  # I wonder why this exists
+        return self
+
+    typing = disnake.abc.Messageable.typing
+    trigger_typing = disnake.abc.Messageable.trigger_typing
```
Comment (onerandomusername) [inline disnake/channel.py:1960]: While these are supported, they should have better docstrings than what they currently do by just being reassigned. I'll update this.

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/channel.py (hunk 6 above)
Comment (Victorsitou) [reply disnake/channel.py:2153]: As we discussed, we will keep it as thread https://canary.discord.com/channels/927603005627437097/927603006579568723/962117732264988682

File: disnake/channel.py (hunk 8 above)
Comment (Victorsitou) [reply disnake/channel.py:2239]: This is probably a topic for a different PR

File: disnake/channel.py (hunk 5 above)
Comment (Victorsitou) [reply disnake/channel.py]: 3bd081fffbbbeb98223139ae0e743ee1d2ef6659

File: disnake/types/threads.py [hunk 10]
```diff
@@ -69,9 +69,18 @@ class Thread(_ThreadOptional):
     message_count: int
     rate_limit_per_user: int
     thread_metadata: ThreadMetadata
+    flags: int
+    applied_tags: SnowflakeList
```
Comment (shiftinv) [inline disnake/types/threads.py:73]: These are both optional, and `applied_tags` appears to be nullable as well (should also have a warning/TODO there)

File: disnake/http.py [hunk 11]
```diff
@@ -1175,6 +1177,83 @@ def get_thread_members(self, channel_id: Snowflake) -> Response[List[threads.Thr
         route = Route("GET", "/channels/{channel_id}/thread-members", channel_id=channel_id)
         return self.request(route)
 
+    def create_tag(
```
Comment (shiftinv) [inline disnake/http.py:1180]: The tag routes should have a warning that they're not finalized

**Review by shiftinv — CHANGES_REQUESTED** [review]
pretty cool! got a preliminary review, haven't tested it so far though.
Additionally:
- the documentation of `Permissions.send_messages` should probably mention that it gives you the ability to create forum threads as well

File: disnake/channel.py [hunk 12]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+
+            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+    last_thread_id: Optional[:class:`int`]
+        The ID of the last created thread in this channel. It may
+        *not* point to an existing or valid thread.
+    default_auto_archive_duration: :class:`int`
+        The default auto archive duration in minutes for threads created in this channel.
+    """
+
+    # TODO: Refer threads as posts?
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py]: this is outdated 😁

File: disnake/channel.py [hunk 13]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: str = ...,
+        topic: Optional[str] = ...,
+        position: int = ...,
+        nsfw: bool = ...,
+        sync_permissions: bool = ...,  # test this
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2028]: Just tested it and it works, needs to be deleted

File: disnake/guild.py [hunk 14]
```diff
@@ -1437,6 +1452,101 @@ async def create_stage_channel(
...
 
+    async def create_forum_channel(
+        self,
+        name: str,
+        *,
+        topic: Optional[str] = None,
+        category: Optional[CategoryChannel] = None,
+        position: int = MISSING,
+        slowmode_delay: int = MISSING,
+        default_auto_archive_duration: AnyThreadArchiveDuration = None,
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/guild.py:1463]: Yes it does work, it seems that aj missed that

File: disnake/channel.py [hunk 15]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+        joined: bool = False,
+        limit: Optional[int] = 50,
+        before: Optional[Union[Snowflake, datetime.datetime]] = None,
+    ) -> ArchivedThreadIterator:
+        """Returns an :class:`~disnake.AsyncIterator` that iterates over all archived threads in the channel.
+
+        You must have :attr:`~Permissions.read_message_history` permission to use this. If iterating over private threads
+        then :attr:`~Permissions.manage_threads` permission is also required.
+
+        .. versionadded:: 2.5
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2293]: Why is that?

File: disnake/channel.py [hunk 16]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+
+    def is_nsfw(self) -> bool:
+        """Whether the channel is marked as NSFW
+
+        :return type: :class:`bool`
+        """
+        return self.nsfw
+
+    @property
+    def last_thread(self) -> Optional[Thread]:
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2001]: Hm, I'm unsure about this. @onerandomusername what do you think?

File: disnake/channel.py [hunk 17]
```diff
@@ -1769,6 +1777,22 @@ def stage_channels(self) -> List[StageChannel]:
         ret.sort(key=lambda c: (c.position, c.id))
         return ret
 
+    @property
+    def forum_channels(self) -> List[ForumChannel]:
+        """List[:class:`ForumChannel`]: Returns the forum channels that are under this category.
+
+        This is sorted by the position and are in UI order from top to bottom.
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:1784]: Oh yeah, but it's mentioned in guild.py 😅

File: disnake/channel.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/channel.py:2293]: The `ForumChannel` type already has a `versionadded`, no need to add a `versionadded` to its methods as well as long as it's teh same version

File: disnake/http.py [hunk 18]
```diff
@@ -1175,6 +1177,83 @@ def get_thread_members(self, channel_id: Snowflake) -> Response[List[threads.Thr
...
+            "name",
+            "auto_archive_duration",
+            "rate_limit_per_user",
+            # Message fields
+            "content",
+            "embeds",
+            "allowed_mentions",
+            "components",
+            "sticker_ids",
+            "attachments",
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/http.py:1240]: Just to make sure, this is handled in `to_multipart_with_attachments` or disnake just doesn't support it? 🤔

File: disnake/http.py (hunk 11 above)
Comment (Victorsitou) [reply disnake/http.py:1180]: What about just deleting them for now? As well as tags types

File: disnake/channel.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2292]: Makes sense 👍

File: disnake/channel.py [hunk 19]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+            You do not have permission to fetch this channel.#
+
+        Returns
+        -------
+        :class:`.Thread`
+            The channel from the ID.
+        """
+        return await self.guild.fetch_channel(thread_id)  # type: ignore
+
+    async def create_thread(
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2155]: Nope, it doesn't seem to support it

File: disnake/threads.py [hunk 20]
```diff
@@ -181,6 +189,7 @@ def _from_data(self, data: ThreadPayload):
         self.slowmode_delay = data.get("rate_limit_per_user", 0)
         self.message_count = data.get("message_count")
         self.member_count = data.get("member_count")
+        self.flags = ChannelFlags._from_value(data.get("flags", 0))
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/threads.py:192]: 8005aa02d0e35142cdc8ab96301e63e12a7a10ed. In my opinion, `Thread._update` should use call `_from_data`, in order to keep everything up to date, would be nice to note it as a suggestion.

File: disnake/threads.py [hunk 21]
```diff
@@ -638,6 +664,8 @@ async def edit(
             payload["invitable"] = invitable
         if slowmode_delay is not MISSING:
             payload["rate_limit_per_user"] = slowmode_delay
+        if pinned is not MISSING:
+            payload["flags"] = 2 if pinned else 0
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/threads.py]: 1a3e8a0de1d920184230bd7f5090c80c93002039

File: disnake/http.py [hunk 22]
```diff
@@ -1175,6 +1177,83 @@ def get_thread_members(self, channel_id: Snowflake) -> Response[List[threads.Thr
...
+        )
+        return self.request(route, json=payload)
+
+    def delete_tag(self, channel_id: Snowflake, tag_id: Snowflake) -> Response[None]:
+        route = Route(
+            "DELETE", "/channels/{channel_id}/tags/{tag_id}", channel_id=channel_id, tag_id=tag_id
+        )
+        return self.request(route)
+
+    def create_forum_thread(
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/http.py]: e3dc0ec55a5281667fd16cd9493c091fed4f7b07

File: disnake/channel.py [hunk 23]
```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
...
+            If no object is passed at all then the defaults given by :attr:`.Client.allowed_mentions`
+            are used instead.
+        view: :class:`.ui.View`
+            A Discord UI View to add to the message. This cannot be mixed with ``components``.
+        components: |components_type|
+            A list of components to include in the message. This cannot be mixed with ``view``.
+        reason: Optional[:class:`str`]
+            The reason for creating the thread. Shows up on the audit log.
+
+        Raises
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2219]: 5c27e0fe273f85130a817b3f9f576af8056eb74a and also raises an error if `files` is more than 10 😄

**Comment by Victorsitou** [comment]
> the documentation of `Permissions.send_messages` should probably mention that it gives you the ability to create forum threads as well

Added in 9626d1ca0f7b513cbf2240a9bd46a2dc8e5a0d33, feel free to suggest a different wording heh

File: disnake/http.py (hunk 11 above)
Comment (shiftinv) [reply disnake/http.py:1180]: Yeah, since they're still wip that would make sense. Maybe keep the code around and add it later, once the API is stable 👍

Comment (onerandomusername) [reply disnake/http.py:1180]: The only tag implementation is in typings and http.py, so I don't think we need to remove them as http.py is not a public facing api

Comment (Victorsitou) [reply disnake/http.py:1180]: > The only tag implementation is in typings and http.py, so I don't think we need to remove them as http.py is not a public facing api

I am aware of this, but tags are "somewhat likely to change", so removing them for now is not a bad idea.  However, I want to keep a branch for specifically that, this PR will only contain basic and documented things

File: disnake/types/threads.py (hunk 10 above)
Comment (Victorsitou) [reply disnake/types/threads.py]: Flags was moved to optional in 1c23993bf854eeb66c4d41228c5e11c5095f763b, also, when is `applied_tags` nullable? From my testing it can be missing/empty list/list of snowflake

File: disnake/types/channel.py [hunk 24]
```diff
@@ -128,13 +128,25 @@ class ThreadChannel(_BaseChannel, _ThreadChannelOptional):
...
 
+class _ForumChannelOptional(TypedDict, total=False):
+    default_auto_archive_duration: ThreadArchiveDurationLiteral
+    available_tags: List[Tag]
+
+
+class ForumChannel(_BaseGuildChannel, _ForumChannelOptional):
+    type: Literal[15]
+    topic: Optional[str]
+    last_message_id: Optional[Snowflake]
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/types/channel.py]: 1c23993bf854eeb66c4d41228c5e11c5095f763b

File: disnake/http.py (hunk 11 above)
Comment (Victorsitou) [reply disnake/http.py]: Tags were removed in aa9055b7cbd5715b831383ce5112f98ed3623f20 😄

File: disnake/webhook/async_.py [hunk 25]
```diff
@@ -543,11 +552,13 @@ def handle_message_parameters(
     elif previous_allowed_mentions is not None:
         payload["allowed_mentions"] = previous_allowed_mentions.to_dict()
 
+    if stickers is not MISSING:
+        payload["stickers"] = [s.id for s in stickers]
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/webhook/async_.py:556]: I agree

[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/webhook/async_.py:556]: That is needed for `stickers` parameter in `create_thread`

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/webhook/async_.py:556]: Ah, because we're using the method from the webhook file. We can refactor that later.

File: disnake/channel.py (hunk 19 above)
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/channel.py:2173]: Closing this at it isn't supported

File: disnake/channel.py (hunk 9 above)
Comment (Victorsitou) [reply disnake/channel.py]: This was changed in b63f5836cb8cfbac17c2f67b9cae391439572b50, thanks arl

File: disnake/webhook/async_.py (hunk 25 above)
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/webhook/async_.py:556]: Closing as it seems to be resolved 😄

File: disnake/channel.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/channel.py:2002]: Let's keep last_thread but remove fetch_thread, as it is just a shortcut to the guild's method.

File: disnake/channel.py [hunk 26]
```diff
@@ -2285,6 +2285,7 @@ async def create_thread(
                 name=name,
                 auto_archive_duration=auto_archive_duration or self.default_auto_archive_duration,
                 rate_limit_per_user=slowmode_delay or 0,
+                type=ChannelType.public_thread.value,
```
Comment (onerandomusername) [inline disnake/channel.py:2288]: why was this added? type cannot be private in a forum channel.

File: disnake/threads.py [hunk 27]
```diff
@@ -94,7 +95,7 @@ class Thread(Messageable, Hashable):
     id: :class:`int`
         The thread ID.
     parent_id: :class:`int`
-        The parent :class:`TextChannel` ID this thread belongs to.
+        The parent Union[:class:`TextChannel`, :class:`ForumChannel`] ID this thread belongs to.
```
Comment (Victorsitou) [inline disnake/threads.py:98]:
```suggestion
        The parent :class:`TextChannel` or :class:`ForumChannel` ID this thread belongs to.
```

File: disnake/channel.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/channel.py:2005]: Then again, get_thread exists on text channels, so I don't really know. Perhaps we should keep it, but it would be harder to remove.

Given the bugs existant with thread caching right now, we should probably not implement this method for now, as it would be easier to add it than to remove it.

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/channel.py:2005]: https://github.com/DisnakeDev/disnake/pull/448/commits/4de2079abe4602f9dd2df72168d0230a23a8de69

File: disnake/channel.py (hunk 26 above)
Comment (onerandomusername) [reply disnake/channel.py:2261]: okay, I understand now, and this makes sense. However, I'm not sure we should be specifying it, but I also don't believe it matters.

File: disnake/threads.py [hunk 28]
```diff
@@ -129,6 +130,12 @@ class Thread(Messageable, Hashable):
         This is only available for threads created after 2022-01-09.
 
         .. versionadded:: 2.4
+
+    flags: :class:`ChannelFlags`
+        The flags the thread has.
+
+        .. versionadded:: 2.5
+
```
Comment (Victorsitou) [inline disnake/threads.py:138]:
```suggestion
```

File: disnake/channel.py [hunk 29]
```diff
@@ -1817,6 +1854,461 @@ class NewsChannel(TextChannel):
...
+        limit: Optional[int] = 50,
+        before: Optional[Union[Snowflake, datetime.datetime]] = None,
+    ) -> ArchivedThreadIterator:
+        """Returns an :class:`~disnake.AsyncIterator` that iterates over all archived threads in the channel.
+
+        You must have :attr:`~Permissions.read_message_history` permission to use this.
+
+        Parameters
+        ----------
+        limit: Optional[:class:`bool`]
```
Comment (Victorsitou) [inline disnake/channel.py:2288]:
```suggestion
        limit: Optional[:class:`int`]
```
This is also wrong in TextChannel

**Review by Victorsitou — COMMENTED** [review]
Other than that lgtm

File: disnake/channel.py (hunk 29 above)
Comment (onerandomusername) [reply disnake/channel.py:2288]: we'll update the ForumChannel docstring in this pr, @shiftinv, could you update it in #470?

**Review by onerandomusername — APPROVED** [review]
looks good, I think this is ready.

**Review by shiftinv — APPROVED** [review]
